### DISCUSSION TITLE
How to execute code after item validation?

### DISCUSSION THREAD

**Discussion by vdegenne** [ibhagwan/fzf-lua#1474] [Q&A] [answered]
I am trying to migrate from telescope to fzf-lua, in my telescope config file I have this:
```lua
    opts = function(_, opts)
      local action_set = require("telescope.actions.set")
      local builtins = require("telescope.builtin")

      local picker_config = {}
      for builtin, _ in pairs(builtins) do
        picker_config[builtin] = {
          -- Execute `zt` after selecting an entry from search results
          attach_mappings = function(prompt_bufnr, _)
            action_set.select:enhance({
              post = function()
                vim.cmd(":normal! zt")
              end,
            })

            return true
          end,
        }
      end

      opts.pickers = vim.tbl_deep_extend("force", picker_config, {})
    end,
```

which will execute `zt` after an item is selected, to have a clear view of all the code below the reached location.

I wonder if it's possible to do that in fzf-lua too.

**Comment by ibhagwan** [comment, marked as answer]
See the advanced Wiki, use the actions table to set the `enter` (or default) action and execute any code you want inside, you can then also call fzf-lua’s `actions.file_edit` or any of the built in actions.

**Reply by vdegenne** [reply]
This is so neat, thanks!
